proposals/nnnn-mutex.md

```diff
@@ -11,18 +11,11 @@ This proposal introduces a mutual exclusion lock, or a mutex, to the standard li
 
 ## Motivation
 
-Since the dawn of Swift, one of the most difficult issues to resolve has been synchronizing access for core standard library collections such as `Array`, `Dictionary` and `Set`. These types are not concurrent and do not provide any sort of thread safety, and are prone to data races. For example, it is a common pitfall to implement a class that contains a property of type `Dictionary` without proper synchronization ensuring that access to this property is well defined across multiple threads.
+In concurrent programs, protecting shared mutable state is one of the core fundamental problems to ensuring reading and writing data is done in an explainable fashion. Synchronizing access to shared mutable state is not a new problem in Swift however. We've introduced many features to help protect mutable data. Actors are a straightforward solution to protecting your mutable state because it isolates the stored data in its own domain. At any one time, only one execution context will have exclusive access to this data. Multiple contexts cannot race with each other and leave your storage in an inconsistent or broken state. The actor model guarantees that all mutable state stored within it is isolated (except `nonisolated` and `let` storage) and operations that access such data are coupled together.
 
-A simple solution to this problem is to protect this mutable state with a lock. Commonly, the protection would be implemented as a mutex, and ensures that only one execution context at a time has read/write access to the value the lock is protecting. Locks make it easy to protect any sort of value used as shared state. There are hundreds of similar lock implementations in Swift code bases. The following are a few implementations in projects related to Swift:
+Ensuring your mutable state is properly synchronized in a coherent way may be as simple as changing your `class` type to an `actor`. However, there may be cases where it's not so simple to do so. For example, a historically synchronous code base may not be able to adopt features like `async`/`await` for any number of reasons, but couldn't embrace the actor model because of that restriction. Perhaps there is data that is allocated together, but isn't mutually exclusive with each other. Should you just allocate two actors?
```

Comment:
```suggestion
Ensuring your mutable state is properly synchronized in a coherent way may be as simple as changing your `class` type to an `actor`. However, there may be cases where it's not so simple to do so. For example, a historically synchronous code base may not be able to adopt features like `async`/`await` for any number of reasons, and cannot embrace the actor model because of that restriction. Perhaps there is data that is allocated together, but isn't mutually exclusive with each other. Should you just allocate two actors?
```

This section reads a bit weird to be honest, I'd consider dropping it entirely And just say as intro to the below section that...

```suggestion
Not all code may be (or want to) able to adopt actors. Reasons for this can be very varied, for example code may have to execute synchronously without any potential for other tasks interleaving with it. Or the `async` effect introduced on methods may prevent legacy code which cannot use Swift Concurrency from interacting with the protected state.
```

WDYT?

Reply:
Sounds good to me!